### ISSUE TITLE
Allow access to Self-Signed Git Repository

### ISSUE THREAD

**Issue by jifox** [nautobot/nautobot#642]
### User Story

As **Austin** the **Network Automation Engineer**, I want to store data in an on premise Git Repository that have self signed Certificates.

I want to store "config contexts" there.

Because it is not possible ether to ignore the certificate validation or add the CA certificate, I'm not able to use that internal Git repository.

As **P.D.** the **Plugin Developer**, I want to store "export templates" and "jobs" in a Git repository with a self-signed certificate too.
 
Actually trying to Clone or Sync the repository leads to following error:

> _Error while refreshing net-automation/nautobot-data/nautobot-configitems: Cmd('git') failed due to: exit code(128) cmdline: git clone -v https://..... /opt/nautobot/git/... stderr: 'Cloning into '/opt/nautobot/git/...'... fatal: unable to access 'https://.../': server **certificate verification failed**. CAfile: none CRLfile: none '_

 
### Database Changes (Optional)

Tbd  (Flag _ignore certificate validation_ or _Trusted Certificate Storage_)

**Comment by glennmatthews** [comment]
Thanks for the feature request! This seems like a very reasonable additional config option to add either as a per-repository toggle or as a system-wide setting in `nautobot_config.py`.

**Comment by jifox** [comment]
@glennmatthews 

After digging deeper I found out, that setting the environemt variable below will solve that problem. I was able to successfully checkout the local git repository.

`GIT_SSL_NO_VERIFY="1"`

I suggest to add this setting to `development/dev.env` as a commented out parameter

```
# Allow self signed git repositories for config contexts, export templates, ...
# GIT_SSL_NO_VERIFY="1"
```

**Comment by smk4664** [comment]
I will take this, I see we are targeting this for 1.2.0, so I will put in a PR for `next`.

**Comment by glennmatthews** [comment]
Great - thank you!

**Comment by glennmatthews** [comment]
Resolved in `next` by #801.
